src/paulscode/android/mupen64plusae/GalleryActivity.java

```diff
@@ -313,7 +326,14 @@ private void refreshGrid( ConfigFile config )
                 String goodName = config.get( md5, "goodName" );
                 String romPath = config.get( md5, "romPath" );
                 String artPath = config.get( md5, "artPath" );
-                items.add( new GalleryItem( this, md5, goodName, romPath, artPath ) );
+                
+                String useName = goodName;
+                String baseName = config.get( md5, "baseName" );
+                if (baseName != null && baseName.length() != 0)
+                    useName = baseName;
+                if (romNames.containsKey(useName) && romNames.get(useName) == Boolean.TRUE)
+                    useName = goodName;
+                items.add( new GalleryItem( this, md5, useName, romPath, artPath ) );
             }
```

Comment:
What is the purpose of this code?

Also, FYI the current main menu is just a stand-in.  Polishing will be done once @paulscode has built the new main menu.